### PR TITLE
ci: pip: use installed version

### PR DESCRIPTION
**Pull request by blueyed**
Currently fails with:
```
    Uninstalling pip-19.3.1:
      Successfully uninstalled pip-19.3.1
Successfully installed pip-20.0
Traceback (most recent call last):
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/bin/pip", line 8, in <module>
    sys.exit(main())
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/site-packages/pip/_internal/cli/main.py", line 73, in main
    command = create_command(cmd_name, isolated=("--isolated" in cmd_args))
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/site-packages/pip/_internal/commands/__init__.py", line 96, in create_command
    module = importlib.import_module(module_path)
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/lib-python/3/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1003, in _gcd_import
  File "<frozen importlib._bootstrap>", line 980, in _find_and_load
  File "<frozen importlib._bootstrap>", line 964, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 674, in _load_unlocked
  File "<builtin>/frozen importlib._bootstrap_external", line 691, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/site-packages/pip/_internal/commands/install.py", line 24, in <module>
    from pip._internal.cli.req_command import RequirementCommand
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/site-packages/pip/_internal/cli/req_command.py", line 20, in <module>
    from pip._internal.operations.prepare import RequirementPreparer
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/site-packages/pip/_internal/operations/prepare.py", line 16, in <module>
    from pip._internal.distributions import (
  File "/opt/hostedtoolcache/PyPy/3.6.9/x64/site-packages/pip/_internal/distributions/__init__.py", line 1, in <module>
    from pip._internal.distributions.source import SourceDistribution
ImportError: cannot import name 'SourceDistribution'
```

Ref: https://github.com/pytest-dev/pytest/runs/400730404

### CODE REVIEW

**Comment by hugovk** [comment]
It's already been reported to pip: https://github.com/pypa/pip/issues/7620

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -122,7 +122,7 @@ jobs:
         python-version: ${{ matrix.python }}
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
+        python -m pip install --upgrade "pip<20"
```
Comment (hugovk) [inline .github/workflows/main.yml]: I expect 20.0.1 will be out with a fix soonish, how about?

```suggestion
        python -m pip install --upgrade "pip!=20.0.0"
```

Comment (blueyed) [reply .github/workflows/main.yml]: I'm more in favor to pinning it to some exact version instead: we should not get disrupted by issues through it, but rather bump it from time to time when feasible - it's just disruptive if it breaks suddenly.

**Comment by blueyed** [comment]
Amended to just use the installed version.

**Comment by The-Compiler** [comment]
I think upgrading pip is a good idea - and it looks like pip 20.0.1 fixing this is out already.

**Comment by blueyed** [comment]
@The-Compiler 
> I think upgrading pip is a good idea

Why?

**Comment by The-Compiler** [comment]
Often, CI environments do have quite outdated versions of pip (admittedly, I'm not sure about GitHub Actions though) - to the point where dependencies are non-installable or a wrong version is installed. One example that comes to mind is platform / python_version markers; another more recent one is manylinux2014 wheels.

**Comment by blueyed** [comment]
@The-Compiler I see, but as can be seen here our image is fine.

**Comment by blueyed** [comment]
@nicoddemus please approve, or close it.

**Comment by nicoddemus** [comment]
Closing then, as in general is recommended to always upgrade `pip` instead of using an installed version.

Of course this opens up the possibility of a pip upgrade being broken, but this is very very rare and usually fixed in hours, because it usually blows up the world.

**Comment by blueyed** [comment]
Just for reference: I've fixed Travis CI (for 3.5.1 though) two (or three) times now since then, which was caused due to package upgrades.

**Pull request closed without merging**